playbook.md

```diff
@@ -49,4 +49,103 @@ This is a playbook for build cops to help deal with problems with the CI infrast
 
      ```
      kubectl logs -l job-name=${JOBNAME}
-     ```
\ No newline at end of file
+     ```
+
+## NFS Volume Is Out Of Disk Space.
+
+1. Use [stackdriver](https://cloud.google.com/filestore/docs/monitoring-instances)
+   to check the disk usage
+
+1. There are two ways to free up disk space
+
+   1. Delete old directories on the NFS share
+
+   1. Delete and recreate the NFS share
+
+   * Both options are outlined below
+
+### Deleting old directories on the NFS share
+
+1. Start a shell in the [debug worker](https://github.com/kubeflow/testing/blob/master/test-infra/ks_app/components/debug-worker.jsonnet)
+
+   ```
+   kubectl exec -it debug-worker-0 /bin/bash
+   ```
+
+1. Delete old directories
+
+   ```
+   cd /mnt/test-data-volume
+   find -maxdepth 1 -type d ! -path . -mtime +7 -exec rm -rf {} ";"
+   ```
+
+### Deleting and recreating the NFS share
+
+1. Delete the nfs deployment
+
+   ```
+   gcloud --project=kubeflow-ci deployment-manager delete kubeflow-ci-nfs
+   ```
+
+1. Delete the PV and pvc
+
+   ```
+   kubectl delete pv gcfs
+   kubectl delete pvc gcfs-storage 
+   ```
+
+1. Wait for them to be deleted
+
+   * Most likely we will need to override [delete protection](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#storage-object-in-use-protection) because there will be some pods still mounting it
+
+
+   * Dump the yaml
+    
+     ```
+     kubectl get pvc nfs-external -o yaml > /tmp/nfs-external.yaml
+     ```
+
+   * Delete the finalizer `kubernetes.io/pvc-protection` in nfs-external.yaml
+
+     ```
+     ...
+     finalizers:
+     labels:
+     ...
+     ```
+
+     * Make sure you have the field finalizers and its an empty list
+
+   * Update the object
+
+     ```
+     kubectl apply -f /tmp/nfs-external.yaml
```

Comment:
Should we also check in this yaml somewhere?

Reply:
It is checked in. 
https://github.com/kubeflow/testing/blob/master/test-infra/ks_app/components/nfs-external.jsonnet

That's the original spec. The spec is then modified by the controller and the finalizers are added. So we are doing a read-modify-write operation here.